### PR TITLE
Configurable name transformation for environment variables

### PR DESCRIPTION
**Pull request by jirkafajfr**
## Description of changes:
The action now has additional parameter called `name-transformation`. Its allowed options are:

- `uppercase` (default): secret names will be transformed into uppercase. This is how this action historically worked, and is default value.
- `lowercase`: secret names will be transformed into the lowercase.
- `none`: no transformation will be performed on the environment variable names and they'll be kept in the same casing as defined in the `secret-ids` property.

## Example 1
```
- name: Get secrets
  uses: aws-actions/aws-secretsmanager-get-secrets@v2
  with:
    secret-ids: exampleSecretName
    name-transformation: lowercase
 ```
 
will yield environment variable `examplesecretname`

## Example 2
```
- name: Get secrets
  uses: aws-actions/aws-secretsmanager-get-secrets@v2
  with:
    secret-ids: exampleSecretName
    name-transformation: none
 ```
 
will yield environment variable `exampleSecretName`

## CI changes
I've split integration tests into several branches where each tests different parameter options. 

image

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -1,2 +1,8 @@
 export const LIST_SECRETS_MAX_RESULTS = 100;
-export const CLEANUP_NAME = 'SECRETS_LIST_CLEAN_UP';
\ No newline at end of file
+export const CLEANUP_NAME = 'SECRETS_LIST_CLEAN_UP';
+
+export enum NameTransformation {
```
Comment (YuvalShAz) [inline src/constants.ts:4]: Maybe instead of enum to avoid switch case in `transformToValidEnvName` function. Something like this could be good for posterity: 

```
export const NAME_TRANSFORMATION = { 
    "None": x => x, 
    "Uppercase": x => x.toUpperCase(), 
    "Lowercase": x => x.toLowerCase() 
}
```

Then in the `transformToValidEnvName` you just do something like:
```
const transformName = NAME_TRANSFORMATION[nameTransformation];
const transformedName = transformName(secretName);
if (!transformedName) {
   throw new Error(`'${nameTransformation}' is unsupported name transformation.`);
}
return transformedName;
```

Comment (YuvalShAz) [reply src/constants.ts:4]: you would also have to change it to loop over `Object.keys` in `parseNameTransformation`

Comment (jirkafajfr) [reply src/constants.ts]: Thanks,
I modified it a bit, and took the idea with having just the transformation function passed around without explicitly specifying implementation inside of the  `transformToValidEnvName`.

File: src/utils.ts [hunk 2]
```diff
@@ -186,14 +195,17 @@ export function isJSONString(secretValue: string): boolean {
...
     if (secretName.match(/^[0-9]/)){
         secretName = '_'.concat(secretName);
     }
 
     // Remove invalid characters
-    return secretName.replace(/[^a-zA-Z0-9_]/g, '_').toUpperCase()
+    secretName = secretName.replace(/[^a-zA-Z0-9_]/g, '_');
+
+    // Apply the name transformation. When no transformation is defined fallback to the "uppercase" transformation
+    return nameTransformation ? nameTransformation(secretName) : secretName.toUpperCase();
```
Comment (simonmarty) [inline src/utils.ts:208]: Can we set nameTransformation to uppercase earlier in the call stack if not set? (In `index.ts` for example)

**Review by simonmarty — CHANGES_REQUESTED** [review]

File: src/utils.ts (hunk 2 above)
Comment (jirkafajfr) [reply src/utils.ts:208]: The `parseTransformationFunction` returns non-nullable, and the real github action execution never encounters the null value. 
https://github.com/aws-actions/aws-secretsmanager-get-secrets/blob/05c8c02fa716d39f8e54243b824ce1abf69acf4d/src/utils.ts#L257

The reason for `nameTransformation` in every util function being nullable 
https://github.com/aws-actions/aws-secretsmanager-get-secrets/blob/05c8c02fa716d39f8e54243b824ce1abf69acf4d/src/utils.ts#L26

is that we have 46 existing unit tests. Each of these tests expects uppercase transformation. Having a newly introduced transformation as optional argument is a way for me to not modify the contract. If this is bothersome I can remove nullability and go *act* by *act* and append `x => x.toUpperCase()`, but I don't think that will make things nicer.

File: src/index.ts [hunk 3]
```diff
@@ -16,10 +17,11 @@ export async function run(): Promise<void> {
         const client : SecretsManagerClient = new SecretsManagerClient({region: process.env.AWS_DEFAULT_REGION, customUserAgent: "github-action"});
         const secretConfigInputs: string[] = [...new Set(core.getMultilineInput('secret-ids'))];
         const parseJsonSecrets = core.getBooleanInput('parse-json-secrets');
+        const nameTransformation = parseTransformationFunction(core.getInput('name-transformation'));
```
Comment (willtong1234) [inline src/index.ts:20]: I may be completely missing something, but don't we need a null check on the name-transformation input here? Or else parseTransformationFunction throws an error right?

Comment (jirkafajfr) [reply src/index.ts:20]: The result of `getInput(...)` is not nullable (checkout the function definition below). It would need to be `: string | null`

```
/**
 * Gets the value of an input.
 * Unless trimWhitespace is set to false in InputOptions, the value is also trimmed.
 * Returns an empty string if the value is not defined.
 *
 * @param     name     name of the input to get
 * @param     options  optional. See InputOptions.
 * @returns   string
 */
export declare function getInput(name: string, options?: InputOptions): string;
```

It returns `uppercase` when user of the action doesn't specify it ([parameter definition](https://github.com/aws-actions/aws-secretsmanager-get-secrets/blob/feature/name-transformation/action.yml#L15-L18)). 

I actually have [integration tests](https://github.com/aws-actions/aws-secretsmanager-get-secrets/blob/feature/name-transformation/.github/workflows/tests.yml#L99-L119) around this. There is no `name-transformation` in the `with:` section and it still passes the `__integration_tests__/name_transformation/uppercase.test.ts`.

**Review by willtong1234 — APPROVED** [review]

**Review by YuvalShAz — APPROVED** [review]
nit: If we want to cover all the bases with the integ tests it could be useful to add edge cases like blank aliases as well (shouldn't affect the correct-ness of this PR)
